EXILED/Exiled.API/Features/Core/UserSettings/ButtonSetting.cs

```diff
@@ -0,0 +1,80 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ButtonSetting.cs" company="ExMod Team">
+// Copyright (c) ExMod Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Features.Core.UserSettings
+{
+    using System;
+    using System.Diagnostics;
+
+    using Exiled.API.Interfaces;
+    using global::UserSettings.ServerSpecific;
+
+    /// <summary>
+    /// Represents a button setting.
+    /// </summary>
+    public class ButtonSetting : SettingBase, IWrapper<SSButton>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ButtonSetting"/> class.
+        /// </summary>
+        /// <param name="id"><inheritdoc cref="SettingBase.Id"/></param>
+        /// <param name="label"><inheritdoc cref="SettingBase.Label"/></param>
+        /// <param name="buttonText"><inheritdoc cref="Text"/></param>
+        /// <param name="holdTime"><inheritdoc cref="HoldTime"/></param>
+        /// <param name="hintDescription"><inheritdoc cref="SettingBase.HintDescription"/></param>
+        /// <param name="header"><inheritdoc cref="SettingBase.Header"/></param>
+        /// <param name="onChanged"><inheritdoc cref="SettingBase.OnChanged"/></param>
+        public ButtonSetting(int id, string label, string buttonText, float holdTime = 0.0f, string hintDescription = null, HeaderSetting header = null, Action<Player, SettingBase> onChanged = null)
+            : base(new SSButton(id, label, buttonText, holdTime, hintDescription), header, onChanged)
```

Comment:
Shouldnt this be :this()

Reply:
Why? It's cctor from SettingBase (`SettingBase(ServerSpecificSettingBase, HeaderSetting, Action)`